src/data/quartiles.rs

```diff
@@ -62,6 +62,33 @@ impl Quartiles {
         }
     }
 
+    /// Create a new quartiles struct with pre-calculated values.
+    ///
+    /// - `s`: The array of the original values
+    /// - **returns** The newly created quartiles
+    ///
+    /// ```rust
+    /// use plotters::prelude::*;
+    ///
+    /// let quartiles = Quartiles::new_from_values(&[2, 24, 31, 39, 45]);
+    /// assert_eq!(quartiles.values(), [2.0, 24.0, 31.0, 39.0, 45.0]);
+    /// ```
+    pub fn new_from_values<T: Into<f64> + Copy + PartialOrd>(s: &[T; 5]) -> Self {
+        let s = s.to_owned();
+        assert!(s[0] <= s[1]);
```

Comment:
In general, we don't want to panic even if things are going wrong, since we are a library, we want to avoid this kind of panic.

Reply:
We could either:

1. collect the array into a Vec then sort
2. change the return type to Result<Self, SomeErrorType>
3. leave it

Personally I would choose 3 (perhaps with an assertion string to explain what happened). If I calculate or receive quartile values and they are not in order, I want to know very directly that something is wrong.

1 is my next choice, but it could silently smooth over some logic error the user made earlier.

2 has both the previous benefits, but a big fallback since users now have to handle an error variant for what should be a very simple operation.

What do you think?

Reply:
I don't think we should panic here, since it doesn't deserve that - more importantly, panicking in a library is more like an "unrecoverable error". This is definitely not "an unrecoverable error". In today's Plotters, there are places may panic unexpectedly, but we are constantly working to get rid of them.  

In principle, we don't allow any panic unless there's no way to be correctly handled or we believe panicking is impossible to happen.

I am fine with both 1 and 2, but I am personally prefer 1. Returning result is still an overkill (And panicking in lib is even bad than that). 

Also if you already copied the data, sorting a 5 elements slice is very lightweight.

Reply:
Ok sounds good, I'll implement option 1 then!